example/main.go

```diff
@@ -2,14 +2,31 @@ package main
 
 import (
 	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
 
-	"github.com/getlantern/pacon"
+	"github.com/getlantern/pac"
 )
 
 func main() {
-	pacon.PacOn("localhost:12345/pac")
-	fmt.Println("proxy set, any key to continue...")
+	pac.SetHelperPath(absPath("./pac-cmd"))
+	if runtime.GOOS == "darwin" {
+		pac.SetIconPathOnOSX(absPath("icon.png"))
```

Comment:
Ahh, of course, because you're setting uid on the executable.  This API is weird because I can call `SetPrompt` multiple times with different values and the last value will win once I call `On` or `Off`.  But, after I've called `On` or `Off`, then subsequent calls to `SetPrompt` have no effect.  In fact, even after running the program a 2nd time, calling `SetPrompt` won't have any effect.  That's very confusing.

Perhaps the API should have three functions:

`go
func EnsureHelperToolExtracted(prompt string, icon string) {}

func On(url string)

func Off(url string)
`

`EnsureHelperToolPresent` would do the extraction and setuid if necessary.  `On` and `Off` would just use the existing helper tool or return an error if the helper tool isn't present.  That would make it clear that:

1. Setting up the helper tool is a one-time initialization activity
2. The prompt and icon are used while setting up the helper tool